sql/core/src/main/java/org/apache/spark/sql/vectorized/ArrowColumnVector.java

```diff
@@ -450,13 +439,11 @@ final boolean isNullAt(int rowId) {
     }
 
     @Override
-    final int getArrayLength(int rowId) {
-      return accessor.getInnerValueCountAt(rowId);
-    }
-
-    @Override
-    final int getArrayOffset(int rowId) {
-      return accessor.getOffsetBuffer().getInt(rowId * accessor.OFFSET_WIDTH);
+    final ColumnarArray getArray(int rowId) {
+      int index = rowId * accessor.OFFSET_WIDTH;
+      int start = offsets.getInt(index);
+      int end = offsets.getInt(index + accessor.OFFSET_WIDTH);
```

Comment:
Does this code work when `rowId == arrayLength`? In other words, is the length of `offsetBuffer` equal to `(rowId + 1) * accessor.OFFSET_WIDTH`?
cc: @ueshin

Reply:
I believe it is, by looking at `accessor.getInnerValueCountAt`. cc @BryanCutler for confirmation.

Reply:
yes, the code is correct.  `offsetBuffer` will be sized to at least `(numRows + 1) * accessor.OFFSET_WIDTH`